api/jwt.go

```diff
@@ -3,77 +3,227 @@ package api
 import (
 	"fmt"
 	"net/http"
-	"os"
+	"reflect"
 	"strings"
-	"time"
 
-	"gitlab.com/NebulousLabs/errors"
-
-	"github.com/NebulousLabs/skynet-accounts/database"
 	"github.com/dgrijalva/jwt-go"
+	"github.com/lestrrat/go-jwx/jwk"
+	"gitlab.com/NebulousLabs/errors"
 )
 
-type (
-	// skynetClaims is a helper struct that defines what claims will be used
-	// with our JWT tokens.
-	skynetClaims struct {
-		UserID string `json:"user_id"`
-		Tier   int    `json:"tier"`
+var (
+	// oathkeeperPubKeys is the public RS key exposed by Oathkeeper for JWT
+	// validation. It's available at oathkeeperPubKeyUrl.
+	oathkeeperPubKeys *jwk.Set = nil
 
-		jwt.StandardClaims
-	}
+	// oathkeeperPubKeyUrl is the URL on which we can find the public key.
+	oathkeeperPubKeyUrl = "http://oathkeeper:4456/.well-known/jwks.json"
```

Comment:
niggle: I have some lint enabled that complains here

```suggestion
	oathkeeperPubKeyURL = "http://oathkeeper:4456/.well-known/jwks.json"
```

Reply:
TIL what niggle means 👍

Reply:
@m-cat can you share the lint config? I write stuff like that automatically sometimes and depend on the linter to help me out. But the linter can only do that if it's properly configured.

PS: TIL about niggle as well - it is an awesome word, it applies to so many things in my life! :D

Reply:
> can you share the lint config?
@ro-tex Looks like my IDE is running `go-golint`. I don't think I have a config for it.

Reply:
And indeed there are other instances of this e.g. `isJsonErr -> isJSONErr`.

Minor stuff but at least y'all learned a new word 🙂

Reply:
It seems that `go-golint` is ` golang.org/x/lint/golint`. Added to `make lint`. 👍